libnetwork/agent.go

```diff
@@ -853,66 +853,56 @@ func (n *Network) handleDriverTableEvent(ev events.Event) {
 		return
 	}
 
-	var (
-		etype driverapi.EventType
-		tname string
-		key   string
-		value []byte
-	)
-
 	switch event := ev.(type) {
 	case networkdb.CreateEvent:
-		tname = event.Table
-		key = event.Key
-		value = event.Value
-		etype = driverapi.Create
+		d.EventNotify(driverapi.Create, n.ID(), event.Table, event.Key, event.Value)
 	case networkdb.DeleteEvent:
-		tname = event.Table
-		key = event.Key
-		value = event.Value
-		etype = driverapi.Delete
+		d.EventNotify(driverapi.Delete, n.ID(), event.Table, event.Key, event.Value)
 	case networkdb.UpdateEvent:
-		tname = event.Table
-		key = event.Key
-		value = event.Value
-		etype = driverapi.Delete
+		// TODO(thaJeztah): is it intentional for an "networkdb.UpdateEvent" event to send a "driverapi.Delete" notification?
+		d.EventNotify(driverapi.Delete, n.ID(), event.Table, event.Key, event.Value)
+	default:
+		// TODO(thaJeztah): are there other event-types? Should this log a warning?
```

Comment:
[`events.Event` is an alias for `any`](https://pkg.go.dev/github.com/docker/go-events#Event) and [the event filter is very lenient](https://github.com/moby/moby/blob/2870559b2c23ed4803f10c566d955e33aa22a61c/libnetwork/networkdb/watch.go#L41-L86) so this function could theoretically be called with other event types. On the other hand, silently discarding is not an unreasonable approach for handling events that an event sink doesn't care about. Perhaps split the difference and log at debug level?

Reply:
> Perhaps split the difference and log at debug level?

Works for me! I added a debug log